src/__tests__/parseWithPointers.spec.ts

```diff
@@ -514,4 +514,126 @@ european-cities: &cities
       expect(result.diagnostics).toEqual([]);
     });
   });
+
+  describe('keys sorting', () => {
+    it('does not retain the order of keys by default', () => {
+      const { data } = parseWithPointers(`foo: true
+bar: false
+"1": false
+"0": true
+`);
+
+      expect(Object.keys(data)).toEqual(['0', '1', 'foo', 'bar']);
+    });
+
+    describe('when sortKeys option is set to true', () => {
+      it('retains the initial order of keys', () => {
+        const { data } = parseWithPointers(
+          `foo: true
+bar: false
+"1": false
+"0": true
+`,
+          { sortKeys: true },
```

Comment:
This seems misleading - I would think that this option means it will sort the keys alphabetically or something, not retain ordering of keys. `maintainKeyOrdering` or something?

Reply:
Yeah, you are right. `maintainKeyOrdering` sounds alright or perhaps `preserveKeyOrder` or something

Reply:
`preserveKeyOrder` sounds good to me.